Signal/src/UIApplication+OWS.swift

```diff
@@ -9,8 +9,20 @@ extension UIApplication {
     var frontmostViewController: UIViewController? {
         let window = UIApplication.shared.keyWindow
         var viewController = window!.rootViewController
-        while viewController?.presentedViewController != nil {
-            viewController = viewController?.presentedViewController
+        
+        while true {
+            if let nextViewController = viewController?.presentedViewController {
+                viewController = nextViewController
+            } else if viewController is UINavigationController {
```

Comment:
style nit: conditional casting removes a line and puts Optional checking in logical place.

```
 // ...
} else if let navigationViewController = viewController as? UINavigationController {
    if let nextViewController = navigationController.topViewController {
      // ...
    }
}
```